src/main/java/edu/ucsb/cs156/dining/controllers/ReviewsController.java

```diff
@@ -109,5 +122,62 @@ public Review postReview(
 
         return savedReview;
     }
+    
+    /**
+     * Update a single review
+     * 
+     * @param id        id of the review to update
+     * @param incoming  the new review
+     * @return the updated review object
+     */
+    @Operation(summary= "Update a single review")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PutMapping("/reviewer")
+    public Review updateReview(
+            @Parameter(name="id") @RequestParam Long id,
+            @RequestBody @Valid Review incoming) {
+            
+        Review review = reviewsRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Review.class, id));
+
+        review.setStars(incoming.getStars());
+        review.setReviewText(incoming.getReviewText());
+        review.setStatus("Awaiting Moderation");
+        review.setModId(null);
+        review.setModComments(null);
+        review.setLastEditedDate(LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS));
+
+        reviewsRepository.save(review);
+
+        return review;
+    }
+
+    /**
+     * Moderate a single review
+     * 
+     * @param id        id of the review to moderate
+     * @param incoming  the new review
+     * @return the updated review object
+     */
+    @Operation(summary= "Moderate a single review")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("/moderator")
+    public Review moderateReview(
+            @Parameter(name="id") @RequestParam Long id,
+            @RequestBody @Valid Review incoming) {
+
+
+        Review review = reviewsRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Review.class, id));
+
+        review.setStatus(incoming.getStatus());
```

Comment:
There should probably be some validation on this field—i.e. check whether the value is one of:
"Awaiting Moderation", "Approved", "Rejected", and if it isn't, throw an `IllegalArgumentException`.

Reply:
finished, with additional testing/methods for handling IllegalArgumentException.